modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/KillQueryTest.java

```diff
@@ -166,19 +225,67 @@ else if (SchemaProposeDiscoveryMessage.class.isAssignableFrom(delegate.getClass(
         return cfg;
     }
 
+
+    private void createJoinCache(String cacheName) {
+        CacheConfiguration<Long, Person> ccfg = GridAbstractTest.defaultCacheConfiguration();
+
+        ccfg.setName(cacheName);
+
+        ccfg.setCacheMode(PARTITIONED);
+        ccfg.setBackups(1);
+        ccfg.setWriteSynchronizationMode(FULL_SYNC);
+        ccfg.setSqlFunctionClasses(TestSQLFunctions.class);
+
+        ccfg.setQueryEntities(Collections.singleton(
+            new QueryEntity(Integer.class.getName(), Person.class.getName())
+                .setTableName("PERSON")
+                .setKeyFieldName("rec_id") // PK
+                .addQueryField("rec_id", Integer.class.getName(), null)
+                .addQueryField("id", Integer.class.getName(), null)
+                .addQueryField("lastName", String.class.getName(), null)
+                .setIndexes(Collections.singleton(new QueryIndex("id", true, "idx_" + cacheName)))
+        ));
+
+        grid(0).createCache(ccfg);
+
+        try (IgniteDataStreamer<Object, Object> ds = grid(0).dataStreamer(cacheName)) {
+            for (int recordId = 0; recordId < MAX_ROWS; recordId++) {
+                int intTabIdFK = ThreadLocalRandom.current().nextInt(MAX_ROWS);
+
+                ds.addData(recordId,
+                    new Person(intTabIdFK,
+                        "Name_" + recordId,
+                        "LastName_" + recordId,
+                        42));
+            }
+        }
+    }
+
+
     /** {@inheritDoc} */
     @Override protected void beforeTestsStarted() throws Exception {
         super.beforeTestsStarted();
 
         cntr = 0;
 
+        GridQueryProcessor.idxCls = MockedIndexing.class;
+
         startGrids(NODES_COUNT);
 
-        for (int i = 0; i < MAX_ROWS; ++i) {
-            grid(0).cache(GridAbstractTest.DEFAULT_CACHE_NAME).put(i, i);
+        try (IgniteDataStreamer<Object, Object> ds = grid(0).dataStreamer(GridAbstractTest.DEFAULT_CACHE_NAME)) {
+            ds.allowOverwrite(false);
```

Comment:
`ds.allowOverwrite(false)` it is a default behavior, no need to specify it explicitly.

Reply:
ou, thanks, done.